packages/@aws-cdk/aws-iotevents/test/detector-model.test.ts

```diff
@@ -0,0 +1,327 @@
+    type: 'Test::Resource',
+    properties: {
+      DetectorModelName: detectorModel.detectorModelName,
+    },
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('Test::Resource', {
+    DetectorModelName: { Ref: 'MyDetectorModel559C0B0E' },
+  });
+});
+
+test('can set physical name', () => {
+  const stack = new cdk.Stack();
+
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    detectorModelName: 'test-detector-model',
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnterEvents: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelName: 'test-detector-model',
+  });
+});
+
+test('can set multiple events to State', () => {
+  const stack = new cdk.Stack();
+
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnterEvents: [
+        {
+          eventName: 'test-eventName1',
+          condition: iotevents.Expression.fromString('test-eventCondition'),
+        },
+        {
+          eventName: 'test-eventName2',
+        },
+      ],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelDefinition: {
+      States: [
+        Match.objectLike({
+          OnEnter: {
+            Events: [
+              {
+                EventName: 'test-eventName1',
+                Condition: 'test-eventCondition',
+              },
+              { EventName: 'test-eventName2' },
+            ],
+          },
+        }),
+      ],
+    },
+  });
+});
+
+test('can set role', () => {
+  const stack = new cdk.Stack();
+
+  // WHEN
+  const role = iam.Role.fromRoleArn(stack, 'test-role', 'arn:aws:iam::123456789012:role/ForTest');
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    role,
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnterEvents: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    RoleArn: 'arn:aws:iam::123456789012:role/ForTest',
+  });
+});
+
+test('can import a DetectorModel by detectorModelName', () => {
+  const stack = new cdk.Stack();
+
+  // WHEN
+  const detectorModelName = 'detector-model-name';
+  const detectorModel = iotevents.DetectorModel.fromDetectorModelName(stack, 'ExistingDetectorModel', detectorModelName);
+
+  // THEN
+  expect(detectorModel).toMatchObject({
+    detectorModelName: detectorModelName,
+  });
+});
+
+test('cannot create without condition', () => {
+  const stack = new cdk.Stack();
```

Comment:
I would recommend putting this in a `beforeEach()` - it's repeated in every test.